server-common/src/main/java/org/apache/kafka/server/share/persister/PersisterStateManager.java

```diff
@@ -356,16 +367,48 @@ public void onComplete(ClientResponse response) {
             if (onCompleteCallback != null) {
                 onCompleteCallback.accept(response);
             }
-            if (response != null && response.hasResponse()) {
-                if (isFindCoordinatorResponse(response)) {
+
+            // We don't know if FIND_COORD or actual REQUEST. Let's err on side of request.
```

Comment:
It seems to me that you could change the signature of `onComplete` to take the request information too. In that way, you'd not have to guess which request failed in the situation where there is no response.

Reply:
@AndrewJSchofield Thanks for the review!
No, the signature of `onComplete` cannot be changed as this method is declared in the general `RequestCompletionHandler` interface implemented by RPC handlers.
The `InterBrokerSendThread` periodically calls `SendThread.generateRequests` which encapsulates the request as a `RequestAndCompletionHandler` whose argument for the request handler is `RequestCompletionHandler`. Then in `RequestCompletionHandler.onComplete` is called which only accepts the clientResponse.

Reply:
Not even with lambda gymnatics like https://github.com/apache/kafka/blob/da166b9dc1ba6476223eb19c267bd5e3babf7109/clients/src/main/java/org/apache/kafka/clients/consumer/internals/FetchRequestManager.java#L152 ?

Reply:
OK, no worries. Just asking the obvious question to see if it sparked an idea. While I don't like the guessing of the request type when there's no response, it's harmless I think.